src/Interface.cc

```diff
@@ -53,7 +53,13 @@ namespace ignition
         auto modelUri = _uri.substr(0,
             _uri.find("files", model.UniqueName().size())-1);
         _client.DownloadModel(common::URI(modelUri), result);
-        result += "/" + fileUrl;
+        std::vector<std::string> tokens = ignition::common::split(fileUrl, "/");
```

Comment:
This is being done inside `ParseModelFileUrl`, do we need to do it again here?

I think that if we embed this logic within `joinPaths`, it can save a lot of duplication.